lib/Form/DataTransformer/DateIntervalToArrayTransformer.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * This file is part of the eZ RepositoryForms package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\RepositoryForms\Form\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * Data transformer from PHP DateInterval to array for form inputs
+ */
+class DateIntervalToArrayTransformer implements DataTransformerInterface
+{
+    private $pad;
+
+    private $fields;
+
+    /**
+     * Constructor.
+     *
+     * @param array  $fields         The date fields
+     * @param bool   $pad            Whether to use padding
+     */
+    public function __construct(array $fields = null, $pad = false)
+    {
+        if ( null === $fields )
+        {
+            $fields = array( 'year', 'month', 'day', 'hour', 'minute', 'second' );
+        }
+
+        $this->fields = $fields;
+        $this->pad = (bool)$pad;
+    }
+
+    /**
+     * Transforms a date interval into an array of date interval elements
+     *
+     * @param \DateInterval $dateInterval Date interval.
+     *
+     * @return array Date interval elements.
+     *
+     * @throws TransformationFailedException If the given value is not an instance of \DateInterval
+     */
+    public function transform( $dateInterval )
+    {
+        if ( null === $dateInterval )
+        {
+            return array_intersect_key(
+                array(
+                    'year' => '',
+                    'month' => '',
+                    'day' => '',
+                    'hour' => '',
+                    'minute' => '',
+                    'second' => '',
+                ),
+                array_flip( $this->fields )
+            );
```

Comment:
Maybe use `+` instead? Besides, instead of using `array_flip()`, maybe `$this->fields` should be set as a has, with field name as key, and e.g. `null` as default value?

It would look like:

```php
if (null === $dateInterval) {
    return $this->fields + [
        'year' => '',
        'month' => '',
        'day' => '',
        'hour' => '',
        'minute' => '',
        'second' => '',
    ];
}
```